test/utf8_test.dart

```diff
@@ -59,21 +71,21 @@ void main() {
   test('fromUtf8 ASCII with length', () {
     final Pointer<Utf8> utf8 = _bytesFromList(
         [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33, 10, 0]).cast();
-    final String end = utf8.toDartString(length: 5);
+    final String? end = utf8.toDartString(length: 5);
```

Comment:
The fact that this type changes to a nullable string is not what we want for users who know that they are not passing in `nullptr`.

Reply:
That's not a problem at all. Those who think they know their pointer to string is never `nullptr` can encode that knowledge in the Dart code as follows:
```
final String end = utf8.toDartString(length: 5)!;
```

`!` is always a good indication for a reviewer to double-check. Whereas `final String end = utf8.toDartString(length: 5);` makes a false impression of this code being ok.